DS-5310 Send user to cancel if they dont want to agree to the consent (when it is enforced)

OH17L opened the pull request:
You can see HTT [here](https://github.com/goalgorilla/open_social/pull/858).

ohorbatiuk committed:
DS-5310 by chmez: Open access to pages which need for deleting user when a user does not want to give consent on a current version of data policy.

ohorbatiuk committed:
DS-5310 by chmez: Use injected service for agreement form.

ohorbatiuk committed:
DS-5310 by chmez: Move data policy settings page.

ohorbatiuk committed:
DS-5310 by chmez: Fix dependency for agreement form.

ohorbatiuk committed:
DS-5310 by chmez: Add message about cancel account on agreement form.

bramtenhove reviewed (changes requested):
See remarks in https://github.com/goalgorilla/open_social/pull/858#pullrequestreview-116300872

ohorbatiuk committed:
Merge branch 'develop' of github.com:goalgorilla/gdpr_consent into feature/DS-5310-delete-user-without-consent

ohorbatiuk committed:
DS-5310 by chmez: Prohibit access to user overview page when enforced consent is required and the user did not give consent on the current version of data policy.

ohorbatiuk committed:
DS-5310 by chmez: Redirect to the Cancel page from agreement page when user press "Save" but checkbox unchecked.

ohorbatiuk committed:
DS-5310 by chmez:

1. Update text of information message on the agreement page.

2. Prohibit access to user edit page when enforced consent is required.

ohorbatiuk committed:
DS-5310 by chmez:

1. Use CKEditor for data policy field.

2. Move information message before the checkbox.

ohorbatiuk committed:
DS-5310 by chmez: Hide information message on agreement page when enforce consent is optional.

bramtenhove committed:
DS-5310 by bramtenhove: Make sure checkbox is only required for signup and when it is enforced
